tests/unit/s2n_certificate_test.c

```diff
@@ -276,6 +299,171 @@ int main(int argc, char **argv)
+        S2N_BLOB_FROM_HEX(basic_constraints_blob, "30 06 01 01 FF 02 01 00");
+        S2N_BLOB_FROM_HEX(key_usage_blob, "03 02 01 86");
+        S2N_BLOB_FROM_HEX(custom_oid_1_blob, "0C 41 6B 65 79 69 64 3A 33 \
+                                              36 3A 36 31 3A 33 46 3A 31 42 3A 30 32 3A 43 37 \
+                                              3A 31 32 3A 32 42 3A 35 33 3A 30 41 3A 32 32 3A \
+                                              42 41 3A 35 38 3A 42 36 3A 41 38 3A 38 30 3A 31 \
+                                              39 3A 45 45 3A 35 31 3A 38 35");
+        S2N_BLOB_FROM_HEX(custom_oid_2_blob, "0C 18 49 50 20 41 64 64 72 65 73 73 3A 31 32 2E \
+                                              33 34 35 2E 36 37 2E 38 39 30");
+        S2N_BLOB_FROM_HEX(custom_oid_3_blob, "0C 28 44 4E 53 3A 31 32 2E 33 34 35 2E 36 37 2E \
+                                              38 39 30 2E 61 75 74 6F 2E 70 64 78 2E 65 63 32 \
+                                              2E 73 75 62 73 74 72 61 74 65");
+
+        struct {
+        const char *oid_in;
+        const char *expected_utf8_oid;
+        struct s2n_blob expected_oid;
+        struct s2n_blob returned_der;
+        bool critical;
+        } test_cases[] = {
+            {
+                .oid_in = "X509v3 Subject Key Identifier",
+                .expected_oid = subject_key_id_blob,
+                .critical = false
+            },
+            {
+                .oid_in = "X509v3 Authority Key Identifier",
+                .expected_oid = authority_key_id_blob,
+                .critical = false
+            },
+            {
+                .oid_in = "X509v3 Basic Constraints",
+                .expected_oid = basic_constraints_blob,
+                .critical = true
+            },
+            {
+                .oid_in = "X509v3 Key Usage",
+                .expected_oid = key_usage_blob,
+                .critical = true
+            },
+            {
+                .oid_in = "1.2.3.4.5.6.7890.1.2.100.1",
+                .expected_utf8_oid = "keyid:36:61:3F:1B:02:C7:12:2B:53:0A:22:BA:58:B6:A8:80:19:EE:51:85",
+                .expected_oid = custom_oid_1_blob,
+                .critical = false
+            },
+            {
+                .oid_in = "1.2.3.4.5.6.7890.1.2.100.2",
+                .expected_utf8_oid = "IP Address:12.345.67.890",
+                .expected_oid = custom_oid_2_blob,
+                .critical = false 
+            },
+            {   .oid_in = "1.2.3.4.5.6.7890.1.2.100.3",
+                .expected_utf8_oid = "DNS:12.345.67.890.auto.pdx.ec2.substrate",
+                .expected_oid = custom_oid_3_blob,
+                .critical = false 
+            },
+            {   .oid_in = "Invalid X509 extension",
+                .critical = false
+            },
+        };
+      
+        /* Test s2n_get_x509_extension_oid_value */
+        {
+            /* Safety checks */ 
```

Comment:
We need safety checks for when there isn't enough data to write the returned values.